code/modules/spells/spell.dm

```diff
@@ -199,12 +199,35 @@
 				return FALSE
 
 	else
-		// If the spell requires wizard equipment and we're not a human (can't wear robes or hats), that's just a given
-		if(spell_requirements & (SPELL_REQUIRES_WIZARD_GARB|SPELL_REQUIRES_HUMAN))
+		// If you strictly need to be a human, well, goodbye.
+		if(spell_requirements & SPELL_REQUIRES_HUMAN)
 			if(feedback)
 				to_chat(owner, span_warning("[src] can only be cast by humans!"))
 			return FALSE
 
+		// Otherwise, we can do some funny snowflake checks in case they're a hat-wearing simple or basic mob.
+		if(spell_requirements & SPELL_REQUIRES_WIZARD_GARB)
```

Comment:
This doesn't account for Ian wearing a wizard hat but perhaps that is a fix for another day.
That said, does this work more generically if you just look for a wizard hat in their contents?